ui/src/components/layout/FlowCanvas.tsx

```diff
@@ -228,7 +216,45 @@ const FlowCanvas: React.FC<Props> = () => {
     ]);
   }
 
-  const onNodeEdit = (event: SyntheticEvent, node: Node) => { console.log(event, node, "Hiyii!")}
+  /**
+   * Edit a Node when it's double clicked
+   */
+   const onNodeEdit = (event: SyntheticEvent, node: Node) => {
+    const vfNode: DisplayNode = store.getById(node.id);
+    const vfType = getAlmostLastPart(vfNode.vfPath);
+    if(vfType == 'process') {
+      const vfProcess = store.getCursor(vfNode.vfPath) as Process;
+      setProcessState(vfProcess);
+      setSelectedDisplayNode(vfNode.id);
+      setType('updateProcess');
+      openModal();
+    }
+  }
+
+  /**
+   * Updates the DisplayNode object after an edit
+   */
+  const afterProcessEdit = (process: Process) => {
+    const displayNode: DisplayNode = store.getById(selectedDisplayNode);
+    displayNode.name = process.name;
+    const newNode = new DisplayNode(displayNode);
+
+    setProcessState(null);
+    setSelectedDisplayNode(null);
+    setType(null);
+
+    setNodes((ns) => {
+      const i = ns.findIndex((node) => node.id === displayNode.id);
+      const nsClone = ns.slice();
```

Comment:
can you explain the deal re: cloning again plz?

Reply:
Here's what they are doing internally, They are also "cloning", but in a round about way. Here they `filter` the array and add back the changes: https://github.com/wbkd/react-flow/blob/ea6247f4a14bc24f4f21d4272de51c20aa73e083/src/utils/changes.ts#L52

I, for some reason, couldn't remember `filter`. I can make this take one fewer line by using `filter` instead of clone.

React internally uses object identity to verify if any changes have taken place, so we need to make sure we pass in a new object so that it actually re-renders our nodes: https://blog.bitsrc.io/understanding-referential-equality-in-react-a8fb3769be0